packages/frontend/src/components/wallet-migration/modals/CleanKeysModal/CleanKeysModal.jsx

```diff
@@ -1,53 +1,255 @@
+                accounts.map((accountId) => getAccountDetails({
+                    accountId,
+                    publicKeyBlacklist: rotatedPublicKeys,
+                    wallet,
+                }))
+            );
+
+            localDispatch({
+                type: ACTIONS.ADD_ACCOUNTS,
+                accounts: accountDetails.reduce(
+                    (acc, account) => {
+                        const { keyType, accountBalance: { balanceAvailable } } = account;
+                        if (keyType === WalletClass.KEY_TYPES.FAK && balanceAvailable >= MINIMUM_ACCOUNT_BALANCE) {
+                            acc.push({ ...account, status: account.accessKeys.length ? null : IMPORT_STATUS.SUCCESS });
+                        }
+
+                        return acc;
+                    },
+                    []
+                ),
+            });
+            setLoadingAccounts(false);
+        };
+        setLoadingAccounts(true);
+        importAccounts();
+    }, [rotatedKeys]);
+
+    const currentAccount = useMemo(
+        () =>  state.accounts.find(({ status }) => status === IMPORT_STATUS.PENDING),
+        [ state.accounts]
+    );
+    const currentFailedAccount = useMemo(
+        () => state.accounts.find(({ status }) => status === IMPORT_STATUS.FAILED)
+            && state.accounts.every(({ status }) => status !== IMPORT_STATUS.PENDING),
+        [ state.accounts]
+    );
+    const batchCleanKeysNotStarted = useMemo(
+        () => state.accounts.every(({ status }) => status === null),
+        [state.accounts]
+    );
+    const completedWithSuccess = useMemo(() => {
+        return !loadingAccounts
+            && state.accounts[state.accounts.length - 1].status !==  IMPORT_STATUS.FAILED
+            && state.accounts.every(({ status }) => status === IMPORT_STATUS.SUCCESS || status === IMPORT_STATUS.FAILED);
+    } , [state.accounts, loadingAccounts]);
+
+    useEffect(() => {
+        if (batchCleanKeysNotStarted) {
+            initialAccountId.current = initialAccountIdOnStart;
+        }
+    },[initialAccountIdOnStart, batchCleanKeysNotStarted]);
+
+
+    useEffect(() => {
+        if (completedWithSuccess) {
+            onNext();
+        }
+    }, [completedWithSuccess]);
+
+    const removeKeysForCurrentAccount = async () => {
+        dispatch(switchAccount({ accountId: currentAccount.accountId }));
+        await new Promise((r) => setTimeout(r, 1500));
+        setShowConfirmCleanupModal(true);
+    };
+
+    useEffect(() => {
+        if (currentAccount) {
+            setShowConfirmCleanupModal(false);
+            setKeysToRemove(currentAccount.accessKeys.reduce((keys, { publicKey }) => {
+                keys[publicKey] = true;
+                return keys;
+            }, {}));
+
+            removeKeysForCurrentAccount();
+        }
+    }, [currentAccount]);
 
     return (
         <MigrationModal>
             <Container>
                 <IconBackground>
                     <IconLogout />
```

Comment:
Need to use different icon. (currently it's logout but it should be key looking icon